code/datums/status_effects/xeno_buffs.dm

```diff
@@ -24,6 +27,335 @@
+	else
+		heal_owner = link_owner
+		heal_target = link_target
+
+	new /obj/effect/temp_visual/telekinesis(get_turf(heal_target))
+	var/heal_amount = round(clamp(amount * (DRONE_ESSENCE_LINK_SHARED_HEAL * stacks), -heal_target.maxHealth, 0))
+	var/heal_remainder = round(max(0, heal_amount - heal_target.getBruteLoss()))
+	heal_target.adjustBruteLoss(heal_amount)
+	heal_target.adjustFireLoss(-heal_remainder)
+	heal_target.adjust_sunder(heal_amount/20)
+	if(heal_amount < 0 || heal_remainder < 0)
+		heal_target.visible_message(span_xenowarning("[heal_target]'s wounds are mended by faint energies."), \
+			span_xenonotice("Through the essence link, [heal_owner] has shared [abs(heal_amount + heal_remainder)] health restoration."))
+
+// Toggles the link signals on or off.
+/datum/status_effect/stacking/essence_link/proc/toggle_link(toggle)
+	if(!toggle)
+		UnregisterSignal(link_owner, list(COMSIG_XENOMORPH_RESIN_JELLY_APPLIED, COMSIG_XENOMORPH_BRUTE_DAMAGE, COMSIG_XENOMORPH_BURN_DAMAGE))
+		UnregisterSignal(link_target, list(COMSIG_XENOMORPH_RESIN_JELLY_APPLIED, COMSIG_XENOMORPH_BRUTE_DAMAGE, COMSIG_XENOMORPH_BURN_DAMAGE))
+		toggle_beam(FALSE)
+		return
+	RegisterSignal(link_owner, COMSIG_XENOMORPH_RESIN_JELLY_APPLIED, .proc/share_jelly)
+	RegisterSignal(link_target, COMSIG_XENOMORPH_RESIN_JELLY_APPLIED, .proc/share_jelly)
+	RegisterSignal(link_owner, COMSIG_XENOMORPH_BRUTE_DAMAGE, .proc/share_heal)
+	RegisterSignal(link_target, COMSIG_XENOMORPH_BRUTE_DAMAGE, .proc/share_heal)
+	RegisterSignal(link_owner, COMSIG_XENOMORPH_BURN_DAMAGE, .proc/share_heal)
+	RegisterSignal(link_target, COMSIG_XENOMORPH_BURN_DAMAGE, .proc/share_heal)
+	toggle_beam(TRUE)
+
+/// Toggles the effect beam on or off.
+/datum/status_effect/stacking/essence_link/proc/toggle_beam(toggle)
+	if(!toggle)
+		QDEL_NULL(current_beam)
+		return
+	current_beam = link_owner.beam(link_target, icon_state= "medbeam", beam_type = /obj/effect/ebeam/essence_link)
+	update_beam()
+
+/// Updates the link's appearance.
+/datum/status_effect/stacking/essence_link/proc/update_beam()
+	var/beam_alpha = round(255 / (max_stacks+1 - stacks)) // 255 is the maximum value possible. We divide that by the missing amount of stacks.
+	if(current_beam)
+		current_beam.visuals.alpha = beam_alpha
+
+// ***************************************
+// *********** Salve Regeneration
+// ***************************************
+/datum/status_effect/salve_regen
+	id = "salve regen"
+	duration = 15 SECONDS
+	tick_interval = 2 SECONDS
+	status_type = STATUS_EFFECT_REFRESH
+	alert_type = null
+
+/datum/status_effect/salve_regen/on_apply()
+	if(!isxeno(owner))
+		return FALSE
+	owner.balloon_alert(owner, "Regenerating")
+	return TRUE
+
+/datum/status_effect/salve_regen/on_remove()
+	owner.balloon_alert(owner, "No longer regenerating")
+	return ..()
+
+/datum/status_effect/salve_regen/tick()
+	var/heal_amount = round(owner.maxHealth * 0.01)
+	var/heal_remainder = max(0, heal_amount - owner.getBruteLoss())
```

Comment:
Uneeded var

Reply:
Donezo.

Reply:
Pretty sure this is necessary, there's no real need to put the math in the proc instead.

Reply:
Meh, whatever.